src/parsers/cdm_parser.jl

```diff
@@ -736,6 +699,71 @@ function services_csv_parser(rsv_raw::DataFrames.DataFrame,gen_raw::DataFrames.D
     return services_dict
 end
 
+"""Return a dictionary that includes a vector of parsed forecast dictionaries."""
+function forecast_csv_parser(ps_dict::Dict, data::Dict)
+    forecasts = Dict(
+        "forecasts" => Vector{Dict}(),
+        "loads" => Vector{Dict}(),
+    )
+
+    # A system can only contain forecasts with one resolution. Skip any that don't
+    # match the first one found.
+    # TODO DT: probably not what we want...
+    first_resolution = nothing
+
+    component_map = retrieve(ps_dict, "name")
+    for key in keys(data)
+        for (name, value) in data[key]
+            if name == "Load"
+                continue
+            end
+            # TODO DT: should we verify that the delta between each timestamp is the same?
+            if value isa DataFrames.AbstractDataFrame
+                len = size(value)[1]
+                @assert len > 2
+                resolution = value[:DateTime][2] - value[:DateTime][1]
+            else
+                len = length(value)
+                @assert len > 2
+                timestamps = TimeSeries.timestamp(value)
+                resolution = timestamps[2] - timestamps[1]
+            end
+
+            component = get(component_map, name, nothing)
+            if isnothing(component)
+                @error "$name is not in component_map, skipping forecast"
```

Comment:
This trips in one case - name = 212_CSP_HEAD_STORAGE.  Seems like there needs to be a secondary lookup to match 212_CSP_1 (both names exist in data/RTS_GMLC/storage.csv).  What do you guys think?

Reply:
I think this is an issue of the RTS data @claytonpbarrows can confirm.